src/hypothesis/searchstrategy/types.py

```diff
@@ -129,6 +135,25 @@ def from_typing_type(thing):
 for t in integer_types:
     _global_type_lookup[t] = st.integers()
 
+if hasattr(os, 'PathLike'):  # pragma: no branch
+    _global_type_lookup[os.PathLike] = \
+        st.fspaths(allow_pathlike=False).map(st._PathLike)
+
+# Also check for PY3 because there is a Python 2 pathlib port on pypi
+# which wont work
```

Comment:
The backported module only works with "str", which works fine on Unix but on Windows you need unicode to support non-ansi paths. But I guess filtering based on the type would work... I'll add that. Only downside is it won't get much testing without the module.